crates/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -366,3 +367,20 @@ fn simplify_signature(
         _ => SimplifyResult::None,
     }
 }
+
+fn simplify_sort(dfg: &mut DataFlowGraph, arguments: &[ValueId]) -> SimplifyResult {
+    match dfg.get_array_constant(arguments[0]) {
+        Some((input, _)) if array_is_constant(dfg, &input) => {
+            let (_, element_type) = dfg.get_numeric_constant_with_type(input[0]).unwrap();
+
+            let mut sorted_inputs: Vec<FieldElement> = vecmap(input, |id| {
+                dfg.get_numeric_constant(id).expect("value id from array should point at constant")
+            });
```

Comment:
`array_is_constant` must go through each array element to call get_numeric_constant on it, then we do the same again on line 377. Can we only have one check for this? E.g. `simplify_sort` can return an `Option` instead then we can use `?` over `expect`.